libs/core/src/lib/select/select.component.ts

```diff
@@ -178,166 +194,320 @@ export class SelectComponent implements ControlValueAccessor, SelectInterface, O
+    }
+
+    /** equal to close
+     * Tells the OptionList if it's closable by simply selecting the item either with CLICK | SPACE | ENTER.
+     * When we are in mobile mode (mobile=true) and have we have enabled confirmation buttons then we
+     * dont close option list. We close only after we hit confirm button.
+     */
+    get canClose(): boolean {
+        return !(this.mobile && this.mobileConfig.approveButtonText);
+    }
+
+    /** Whether control can be interacted with */
+    get isInteractive(): boolean {
+        return !(this.readonly || this.disabled);
+    }
+
+    get isCancelableMobileSelect(): boolean {
+        return this.mobile && !!this.mobileConfig.approveButtonText
+    }
+
+    /**
+   * Check if we can emit a value. They are 3 different paths where we can emit value:
+   *  - We are not in mobile mode [mobile]=false
+   *  - We are in mobile mode [mobile]=true and mobile dialog as just close X button to dismiss it
+   *  - We are in mobile mode [mobile]=true and mobile dialog has ApproveButton available and
+   *  dialog is closed. (meaning we already clicked this button)
+   */
+    get canEmitValueChange(): boolean {
+        return (
+            !this.mobile ||
+            (this.mobile &&
+                // Approve + Cancel
+                !!this.mobileConfig.approveButtonText &&
+                !this._isOpen) ||
+            // CloseButton. Emits when you click on the option item
+            (this.mobileConfig.hasCloseButton && !!this.mobileConfig.approveButtonText === false)
+        );
     }
 
     /** @hidden */
-    @HostListener('window:resize')
-    resizeScrollHandler(): void {
-        this.calculatedMaxHeight = this.mobile ? null : window.innerHeight * 0.45;
+     _compareWith = (o1: any, o2: any) => o1 === o2;
+    /**
+     * Function to compare the option values with the selected values.
+     */
+    get compareWith(): (o1: any, o2: any) => boolean {
+        return this._compareWith;
     }
 
+    set compareWith(fn: (o1: any, o2: any) => boolean) {
+        if (typeof fn !== 'function') {
+            throw Error('compareWith` must be a function.');
+        }
+        this._compareWith = fn;
+        if (this._selectionService._selectionModel) {
+            // A different comparator means the selection could change.
+            this._initializeSelection();
+        }
+    }
+
+    get focused(): boolean {
+        return this._focused || this._isOpen;
+    }
+
+    get calculatedMaxHeight(): number {
+        return this._maxHeight || this._calculatedMaxHeight;
+    }
+
+
+
     constructor(
+        public viewportRuler: ViewportRuler,
         private _elementRef: ElementRef,
-        private _selectProxy: SelectProxy,
+        @Attribute('tabindex') _tabIndex: string,
+        public dir: Directionality,
+        public _KeySerivce: SelectKeyManagerService,
+        public _selectionService: SelectSelectionModelService,
```

Comment:
Could you make them `private`? I can't see any usages outside component and tests

Reply:
It is used in selection service test specs.

Reply:
I see, in spec tests you can use `any`  casting, like [here](https://github.com/SAP/fundamental-ngx/blob/fb48296246fa53d6578f87017351e6dfa91ea47c/libs/core/src/lib/alert/alert.component.spec.ts#L61). Could you apply it?